### PR TITLE
Added event for Meshery design import fail

### LINKED ISSUE #9259: Missing Notification Center Event for failed Application import
### Current Behavior
When I use her imports a Kubernetes Podcast, but Nik likes to select that as their application source type, import fails. This is expected. 

Upon failure, a toast notification is displayed. This is good, but not enough. 

### Expected Behavior

A Notification Center event needs to be produced and replete with information about how and why the failure occurred and with and error code and what to do to remedy the situation.

### Screenshots/Logs
Screenshot 2023-10-31 at 9 11 25 PM

---
### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/contributing/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and [designs for Meshery UI](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by Yashsharma1911**
**Notes for Reviewers**

This PR fixes #9259 

This PR adds a new error event on failing Meshery design import to make sure it gets persisted in the notification center

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

File: server/handlers/error.go [hunk 1]
```diff
@@ -200,7 +200,7 @@ func ErrPatternFile(err error) error {
 }
 
 func ErrInvalidPattern(err error) error {
-	return errors.New(ErrInvalidPatternCode, errors.Alert, []string{"Invalid design, execution is infeasible", err.Error()}, []string{err.Error()}, []string{"Trait Definition is invalid or unable to process"}, []string{"Check Trait Definition"})
+	return errors.New(ErrInvalidPatternCode, errors.Alert, []string{"Invalid design, execution is infeasible", err.Error()}, []string{err.Error()}, []string{"Services/Traits definition is invalid or unable to process"}, []string{"Ensure that Services/Traits definition present in meshery design", "Ensure that Services/Traits definition is not invalid"})
```
Comment (leecalcote) [inline server/handlers/error.go:203]: @Yashsharma1911 thanks. We don't use Traits any longer and will hopefully get rid of Services, too. @MUzairS15

File: server/handlers/meshery_pattern_handler.go [hunk 2]
```diff
@@ -315,6 +315,11 @@ func (h *Handler) handlePatternPOST(
 				h.log.Error(ErrInvalidPattern(err))
 				http.Error(rw, ErrInvalidPattern(err).Error(), http.StatusBadRequest)
 				addMeshkitErr(&res, ErrInvalidPattern(err))
+				event := eventBuilder.WithSeverity(events.Error).WithMetadata(map[string]interface{}{
+					"error": ErrInvalidPattern(err),
+				}).WithDescription("invalid design file format").Build()
```
Comment (MUzairS15) [inline server/handlers/meshery_pattern_handler.go:320]:
```suggestion
				}).WithDescription("Invalid design file format").Build()
```

File: server/handlers/error.go (hunk 1 above)
Comment (MUzairS15) [inline server/handlers/error.go:203]: Instead of execution is infeasible, please add the operation name itself. Import/Update/…

Comment (leecalcote) [reply server/handlers/error.go:203]: @Yashsharma1911 ....

Comment (Yashsharma1911) [reply server/handlers/error.go:203]: Removing traits from suggestion

Comment (Yashsharma1911) [reply server/handlers/error.go:203]: Make sense updating

Comment (leecalcote) [reply server/handlers/error.go:203]: Always capitalize Meshery, @Yashsharma1911

**Comment by leecalcote** [comment]
Syncing with latest on master.

**Comment by MUzairS15** [comment]
@Yashsharma1911 Were are we heading on this?

**Comment by leecalcote** [comment]
@Yashsharma1911 ping, ping.

**Comment by leecalcote** [comment]
puh-ping

**Comment by Yashsharma1911** [comment]
I accidentally unsubscribed to this PR 😬 was not getting notifications, I'll pushing the requested changes, thanks for calling on this and giving time

**Comment by Yashsharma1911** [comment]
Updated the error message, I also added this link https://docs.meshery.io/concepts/designs for people to understand design better however we are not doing a good job on that page, I'll update the doc there

**Comment by Yashsharma1911** [comment]
Created this issue to track https://github.com/meshery/meshery/issues/9751

File: server/handlers/error.go [hunk 3]
```diff
@@ -187,7 +187,7 @@ func ErrPatternFile(err error) error {
 }
 
 func ErrInvalidPattern(err error) error {
-	return errors.New(ErrInvalidPatternCode, errors.Alert, []string{"Invalid design, execution is infeasible", err.Error()}, []string{err.Error()}, []string{"Trait Definition is invalid or unable to process"}, []string{"Check Trait Definition"})
+	return errors.New(ErrInvalidPatternCode, errors.Alert, []string{"Invalid design, unable to import design", err.Error()}, []string{err.Error()}, []string{"The services definition in your Meshery design is either missing or invalid"}, []string{"Please review your Meshery design to ensure that it includes valid services definitions.", "If you are unfamiliar with Meshery design, consider referring to: https://docs.meshery.io/concepts/designs for guidance."})
```
Comment (MUzairS15) [inline server/handlers/error.go:190]: @Yashsharma1911 The `probable cause` and `suggested remediation` should be in pairs i.e. for each cause there is a corresponding remedy. In the UI as well we display as 1-1 mapping b/w these two.

File: server/handlers/meshery_pattern_handler.go [hunk 4]
```diff
@@ -313,6 +313,11 @@ func (h *Handler) handlePatternPOST(
 				h.log.Error(ErrInvalidPattern(err))
 				http.Error(rw, ErrInvalidPattern(err).Error(), http.StatusBadRequest)
 				addMeshkitErr(&res, ErrInvalidPattern(err))
+				event := eventBuilder.WithSeverity(events.Error).WithMetadata(map[string]interface{}{
+					"error": ErrInvalidPattern(err),
+				}).WithDescription("Invalid design file format").Build()
```
Comment (MUzairS15) [inline server/handlers/meshery_pattern_handler.go:318]: Please publish the event as well.

**Review by MUzairS15 — CHANGES_REQUESTED** [review]

**Comment by leecalcote** [comment]
Thanks for reviewing this, @MUzairS15. Some feedback for @Yashsharma1911

**Comment by leecalcote** [comment]
@Yashsharma1911 merge conflict.

File: server/handlers/error.go (hunk 1 above)
Comment (leecalcote) [reply server/handlers/error.go]: This has resurfaced recently, @MUzairS15

File: server/handlers/error.go (hunk 3 above)
Comment (leecalcote) [reply server/handlers/error.go:190]: @Althaf66 @RipulHandoo it's good for both of you to be aware of this formatting.

**Comment by leecalcote** [comment]
@Althaf66 perhaps you could assist with this merge conflict.

**Comment by RipulHandoo** [comment]
> @Althaf66 perhaps you could assist with this merge conflict.

@leecalcote merge conflict resolved. ✅

**Comment by leecalcote** [comment]
Thank you, @RipulHandoo

**Comment by leecalcote** [comment]
@MUzairS15, @RipulHandoo might need additional privileges in order to push the commit for eliminated merge conflict.

**Comment by leecalcote** [comment]
Syncing branch with `master`.

**Comment by leecalcote** [comment]
@Yashsharma1911 all good to go?

**Pull request closed without merging**